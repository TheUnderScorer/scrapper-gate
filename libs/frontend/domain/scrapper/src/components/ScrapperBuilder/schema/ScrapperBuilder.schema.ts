import { ScrapperBuilderFormState } from '../ScrapperBuilder.types';
import {
  ScrapperRunSettingsInputSchema,
  VariableInputSchema,
} from '@scrapper-gate/shared/validation';
import joi from 'joi';
import { ScrapperStepNodeSchema } from './ScrapperStepNode.schema';

export const ScrapperBuilderSchema = joi.object<ScrapperBuilderFormState>({
  items: joi.array().items(ScrapperStepNodeSchema),
  name: joi.string().required(),
  variables: joi.array().items(VariableInputSchema).allow(null),
  runSettings: ScrapperRunSettingsInputSchema,
});
