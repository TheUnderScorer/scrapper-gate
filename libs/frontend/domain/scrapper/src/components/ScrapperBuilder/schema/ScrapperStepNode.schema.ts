import { FlowBuilderNodeTypes } from '@scrapper-gate/frontend/flow-builder';
import { ScrapperStepInputSchema } from '@scrapper-gate/shared/validation';
import joi from 'joi';
import { isEdge } from 'react-flow-renderer';
import { ScrapperBuilderNode } from '../ScrapperBuilder.types';

const omitTypes = [FlowBuilderNodeTypes.Start, FlowBuilderNodeTypes.End];

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const whenSchema = joi.object().custom((item: ScrapperBuilderNode, helpers) => {
  if (isEdge(item) || omitTypes.includes(item.type as FlowBuilderNodeTypes)) {
    return helpers.error('any');
  }

  return item;
});
export const ScrapperStepNodeSchema = joi
  .object<Pick<ScrapperBuilderNode, 'data' | 'type'>>({
    data: joi.object(),
    type: joi.string(),
  })
  .when(whenSchema, {
    then: {
      data: ScrapperStepInputSchema,
    },
  });
