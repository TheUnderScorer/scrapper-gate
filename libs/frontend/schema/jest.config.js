module.exports = {
  displayName: 'frontend-schema',
  preset: '../../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../../coverage/libs/frontend/schema',
};
