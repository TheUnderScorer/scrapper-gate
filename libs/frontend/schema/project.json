{
  "root": "libs/frontend/schema",
  "sourceRoot": "libs/frontend/schema/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/frontend/schema/**/*.{ts,tsx,js,jsx}"]
      },
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/frontend/schema"],
      "options": {
        "jestConfig": "libs/frontend/schema/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
