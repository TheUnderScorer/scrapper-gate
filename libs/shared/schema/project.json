{
  "root": "libs/shared/schema",
  "sourceRoot": "libs/shared/schema/src",
  "projectType": "library",
  "targets": {
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/shared/schema/**/*.ts"]
      },
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/shared/schema"],
      "options": {
        "jestConfig": "libs/shared/schema/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}
