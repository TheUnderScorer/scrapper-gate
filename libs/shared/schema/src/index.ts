export * from './typeDefs';
export * from './types';
export * from './query.types';
export * from './scalars/Date.scalar';
export * from './scalars/Record.scalar';
