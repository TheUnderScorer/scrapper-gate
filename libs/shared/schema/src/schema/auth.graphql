# REST - don't remove

input LoginInput {
  username: String!
  password: String!
}

type LoginResponse {
  accessToken: String!
  refreshToken: String!
}

input ForgotPasswordInput {
  username: String!
}

type ForgotPasswordResponse {
  error: String
  stack: String
}

input ResetPasswordInput {
  newPassword: String!
}

type ResetPasswordResponse {
  error: String
  stack: String
}

type IsAutenthicatedResponse {
  isAutenthicated: Boolean
}

# END REST

extend type Mutation {
  #REST don't remove
  login(input: LoginInput!): LoginResponse
  forgotPassword(input: ForgotPasswordInput): ForgotPasswordResponse
  resetPassword(
    input: ResetPasswordInput
    token: String!
  ): ResetPasswordResponse
  #END REST
}

extend type Query {
  isAutenthicated: IsAutenthicatedResponse
}
