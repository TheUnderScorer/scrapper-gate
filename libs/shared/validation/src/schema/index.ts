export * from './conditionalRules/HtmlConditionalRule.schema';
export * from './conditionalRules/VariableConditionalRule.schema';
export * from './conditionalRules/DateRule.schema';
export * from './scrapper/ScrapperInput.schema';
export * from './scrapper/ScrapperStepInput.schema';
export * from './scrapper/StartScrapperInput.schema';
export * from './scrapper/ScrapperConditionalRuleGroupInput.schema';
export * from './scrapper/ScrapperRunnerMessagePayload.schema';
export * from './scrapper/ScrapperRunSettingsInput.schema';
export * from './variables/VariableInput.schema';
export * from './DurationInput.schema';
export * from './LoginInput.schema';
export * from './PositionInput.schema';
export * from './Selector.schema';
export * from './scrapper/CreateScrapperInput.schema';
