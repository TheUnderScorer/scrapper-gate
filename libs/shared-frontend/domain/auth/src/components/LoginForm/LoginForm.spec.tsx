import { LoginDocument } from '@scrapper-gate/shared-frontend/schema';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import {
  LoginMutation,
  LoginMutationVariables,
} from '@scrapper-gate/shared/schema';
import {
  LoginForm,
  LoginFormProps,
} from '@scrapper-gate/shared-frontend/domain/auth';
import { act, render, RenderResult, waitFor } from '@testing-library/react';
import fireEvent from '@testing-library/user-event';
import React from 'react';

const mocks: MockedResponse[] = [
  {
    request: {
      query: LoginDocument,
      variables: {
        input: {
          username: 'test@test.com',
          password: 'password',
        },
      } as LoginMutationVariables,
    },
    result: {
      data: {
        login: {
          accessToken: '#token',
          refreshToken: '#refresh_token',
        },
      } as LoginMutation,
    },
  },
];

const renderComponent = (props?: LoginFormProps): RenderResult => {
  let cmp: RenderResult;

  act(() => {
    cmp = render(
      <MockedProvider mocks={mocks}>
        <LoginForm {...props} />
      </MockedProvider>
    );
  });

  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
  return cmp!;
};

describe('<LoginForm />', () => {
  it('should render without crashing', () => {
    const cmp = renderComponent();
    expect(cmp).toMatchSnapshot();
  });

  it('should handle login', async () => {
    const handleLogin = jest.fn();

    const { container } = renderComponent({
      afterLogin: handleLogin,
    });

    const username = container.querySelector('#username');
    const password = container.querySelector('#password');
    const loginBtn = container.querySelector('#login');

    await act(async () => {
      await fireEvent.type(username, 'test@test.com', {
        delay: 10,
      });

      await fireEvent.type(password, 'password', {
        delay: 10,
      });
    });

    await act(async () => {
      fireEvent.click(loginBtn);
    });

    await waitFor(() => expect(handleLogin).toHaveBeenCalledTimes(1));
  });
});
