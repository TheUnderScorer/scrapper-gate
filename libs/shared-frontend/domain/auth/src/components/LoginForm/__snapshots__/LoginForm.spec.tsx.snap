// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginForm /> should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="makeStyles-form-1"
      >
        <div
          class="css-m72s89-Stack-root"
        >
          <div
            class="css-1a8w37c"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              <span>
                Sign in to 
                <strong>
                  Scrapper Gate
                </strong>
                 
                <img
                  alt="ðŸš€"
                  draggable="false"
                  src="https://twemoji.maxcdn.com/2/svg//1f680.svg"
                  style="height: 1em; width: 1em; margin: 0px .05em 0px .1em; vertical-align: -0.1em;"
                />
              </span>
            </h5>
          </div>
          <div
            style="opacity: 0; visibility: hidden;"
          />
          <div
            class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-88nued-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="username"
              id="username-label"
            >
              E-mail
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-zokxp0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1gpj8ho-MuiInputBase-input-MuiOutlinedInput-input"
                id="username"
                name="username"
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-2s9zix"
                >
                  <span>
                    E-mail
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-88nued-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Password
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-zokxp0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1gpj8ho-MuiInputBase-input-MuiOutlinedInput-input"
                id="password"
                name="password"
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-2s9zix"
                >
                  <span>
                    Password
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-1pxzs7f-MuiButtonBase-root-MuiButton-root"
            id="login"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label css-8xplcm-MuiButton-label"
            >
              Login
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-4nj22a-Stack-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              Don't have account?
            </p>
            <button
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-d0ukip-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label css-8xplcm-MuiButton-label"
              >
                Sign up
              </span>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="makeStyles-form-1"
    >
      <div
        class="css-m72s89-Stack-root"
      >
        <div
          class="css-1a8w37c"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
          >
            <span>
              Sign in to 
              <strong>
                Scrapper Gate
              </strong>
               
              <img
                alt="ðŸš€"
                draggable="false"
                src="https://twemoji.maxcdn.com/2/svg//1f680.svg"
                style="height: 1em; width: 1em; margin: 0px .05em 0px .1em; vertical-align: -0.1em;"
              />
            </span>
          </h5>
        </div>
        <div
          style="opacity: 0; visibility: hidden;"
        />
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-88nued-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="username"
            id="username-label"
          >
            E-mail
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-zokxp0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1gpj8ho-MuiInputBase-input-MuiOutlinedInput-input"
              id="username"
              name="username"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  E-mail
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          class="MuiFormControl-root MuiFormControl-marginNone MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-88nued-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="false"
            for="password"
            id="password-label"
          >
            Password
          </label>
          <div
            class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-sizeSmall css-zokxp0-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1gpj8ho-MuiInputBase-input-MuiOutlinedInput-input"
              id="password"
              name="password"
              type="password"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-2s9zix"
              >
                <span>
                  Password
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root css-1pxzs7f-MuiButtonBase-root-MuiButton-root"
          id="login"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label css-8xplcm-MuiButton-label"
          >
            Login
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <div
          class="css-4nj22a-Stack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
          >
            Don't have account?
          </p>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-d0ukip-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label css-8xplcm-MuiButton-label"
            >
              Sign up
            </span>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
